src/euroleague_api/team_lineup_euroleague_api.py

```diff
@@ -0,0 +1,98 @@
+import pandas as pd
+import numpy as np
+from euroleague_api.play_by_play_data import PlayByPlay
+from euroleague_api.boxscore_data import BoxScoreData
+
+season = 2023
+gamecode = 302
+
+pd.options.mode.chained_assignment = None
+pd.set_option('display.max_columns', None)
+pd.set_option('display.width', 170)
+pd.set_option('display.max_rows', None)
+pd.set_option('display.float_format', lambda x: '%.3f' % x)
+
+def pbp_data_with_lineups(season, game_code):
+    
+    # Getting pbp for the match
+    pbp = PlayByPlay()
+    pbp_data = pbp.get_game_play_by_play_data(season=season, gamecode=game_code)
+    pbp_data['Indices'] = range(0, len(pbp_data)) # resetting indices since going to loop through them, don't want any duplicates
+    pbp_data.set_index('Indices', inplace=True) # resetting indices since going to loop through them, don't want any duplicates
+
+    # Getting starting five
+    boxscoredata = BoxScoreData()
+    game_bxscr_stats = boxscoredata.get_player_boxscore_stats_data(season=season, gamecode=game_code)
+
+    starting_five = game_bxscr_stats.loc[game_bxscr_stats["IsStarter"] == 1, ["Team", "Player"]]
+    starting_five['ID'] = starting_five.groupby('Team').cumcount()
+    starting_five = starting_five.pivot(index='ID', columns='Team', values='Player')
+    starting_five.reset_index(drop=True, inplace=True)
+
+    # Creating player columns
+    on_court_cols = ['Team_A1', 'Team_A2', 'Team_A3', 'Team_A4', 'Team_A5', 'Team_B1', 'Team_B2', 'Team_B3', 'Team_B4', 'Team_B5']
+    pbp_data[on_court_cols] = np.nan
+
+    # Filling start of the match with starters
+    on_court_mask1 = ['Team_A1', 'Team_A2', 'Team_A3', 'Team_A4', 'Team_A5']
+    on_court_mask2 = ['Team_B1', 'Team_B2', 'Team_B3', 'Team_B4', 'Team_B5']
+
+    pbp_data.loc[0, on_court_mask1] = starting_five.iloc[:, 0].tolist()
+    pbp_data.loc[0, on_court_mask2] = starting_five.iloc[:, 1].tolist()
+
+    # Capturing subs and replacing player columns with players who substitute in
+    players = pbp_data.loc[min(pbp_data.index.tolist()), on_court_cols]
```

Comment:
if pbp_data.index is reset as suggested above, this line can be simplified as `players = pbp_data.loc[0, on_court_cols]`